### ISSUE TITLE
simple test of 'readtable.m' gives err msg about 'table'

### ISSUE THREAD

**Issue by fenghaolin** [octave-de/table#1]
I thought the readtable.m was the octave open-source version of the same function in MATLAB (>=2013b). Then I try the following simple example:
`T = readtable('myCSVTable.dat');`
which is from the "help" example of MATLAB. And what I got is:

> error: Invalid call to table.  Correct usage is:
> 
>  -- Function File: [T, L_X] = table (X)
>  -- Function File: [T, L_X, L_Y] = table (X, Y)
> 
> error: called from
>     print_usage at line 90 column 5
>     table at line 35 column 5
>     readtable at line 39 column 11

This is probably because the "table" function used in your readtable.m is different than the one in the octave statistics package. Am I right? And how to fix this? Thanks.

**Comment by markuman** [comment]
Yes, it aims to be the open-source version of Matlab `readtable()` and the `table()` class. But it's completly broken at the moment and I never go on with it.
1. this `readtable()` doesn't support `.dat` file format yet. Only `.csv` or `.xlsx` formats will work.
2. the `table` class itself it not finished yet. I've started `table()` long time ago, but get stuck
   while overloading `subsref` and `subsasgn`. This is absolutly necessary to feel
   comfortable with `table()`.

> And how to fix this?

the `@table` class needs a lot of care, improvments and work. `readtable()` is in the end only a wrapper between file formats and `@table` class.

**Comment by fenghaolin** [comment]
Hi. Thanks for your reply. The `.dat` is only a filename extension. The file itself is just in csv format, and it is plain text. I created it following the matlab help example of its own readtable. So, does it means right now there is no immediate fix about this yet? 
Thanks once again.

**Comment by markuman** [comment]
given this `test.csv` for example

```
date, test head, degree
01.01.1990, 513, 14.3
01.02.1990, 514, 13.9
```

you can do

```
octave:1> a = readtable('test.csv', 'delimiter', ',', 'HeaderLines', 1)
a =

  scalar structure containing the fields:

    Var1 = 
    {
      [1,1] = 01.01.1990
      [2,1] = 01.02.1990
    }
    Var2 = 
    {
      [1,1] =  513
      [2,1] =  514
    }
    Var3 = 
    {
      [1,1] =  14.3
      [2,1] =  13.9
    }
```

but that's all. As I said, it's not ready for serious usage at the moment!

you should better use `csv2cell()` from io package or this one https://github.com/octave-de/macgyver_utils/blob/master/csv2cell.m and avoid using tables in Matlab when your code have to be compatible with octave.

**Comment by fenghaolin** [comment]
Thanks. Looking forward to the complete package of table

**Comment by tpl2go** [comment]
hi, just wondering if this table development for Octave is still active. I recently realized that the table class of MATLAB (http://www.mathworks.com/help/matlab/ref/table.html) is not yet implemented in Octave and am hoping that there is some active development on it.

**Comment by markuman** [comment]
This here is currently dead. Feel free to start table implementation youself.
